src/java.base/share/classes/jdk/internal/classfile/impl/ClassFileImpl.java

```diff
@@ -83,17 +83,28 @@ public ClassFileImpl withOptions(Option... options) {
         var chro = classHierarchyResolverOption;
         var amo = attributeMapperOption;
         for (var o : options) {
-            switch (o) {
-                case StackMapsOption oo -> smo = oo;
-                case DebugElementsOption oo -> deo = oo;
-                case LineNumbersOption oo -> lno = oo;
-                case AttributesProcessingOption oo -> apo = oo;
-                case ConstantPoolSharingOption oo -> cpso = oo;
-                case ShortJumpsOption oo -> sjo = oo;
-                case DeadCodeOption oo -> dco = oo;
-                case DeadLabelsOption oo -> dlo = oo;
-                case ClassHierarchyResolverOption oo -> chro = oo;
-                case AttributeMapperOption oo -> amo = oo;
+            if (o instanceof StackMapsOption oo) {
```

Comment:
Just out of curiosity, will this be slower than the switch statement?

Reply:
It may after `SwitchBootstraps` is improved; but this code will be relied on by `SwitchBootstraps` to generate switches, so it won't be able to use pattern matching switches.

Reply:
I don't think (throughput) performance matters here, based on patterns in JDK internal code. We use either `ClassFile.of()` which retrieves a pre-initialized singleton (creating which doesn't touch this method) or create a private static final like in `ProxyGenerator`:
```
    private static final ClassFile CF_CONTEXT =
            ClassFile.of(ClassFile.StackMapsOption.DROP_STACK_MAPS);
```